### PR TITLE
test(ESD-1386): add validation helper coverage and boundary-value tests

### PR DESCRIPTION
**Pull request by Phil-Browne**
The `internal/validation` shared helpers had no test coverage. Adds tests for the `Get*FromInterface` coercion helpers (used on every JSON input path), the `Validate*` primitives, and `ValidationError` formatting, plus boundary-value cases for ASN, IPv4, CIDR, and port name length.

While writing the IPv4/CIDR edge cases, two loose-validation bugs surfaced and are fixed:

- `ValidateIPv4` accepted IPv4-mapped IPv6 strings like `::ffff:1.2.3.4` because `net.ParseIP(...).To4()` succeeds for them. Now rejects anything containing a colon.
- `ValidateCIDR` accepted IPv6 CIDR (e.g. `2001:db8::/32`). All consumers are IPv4-only contexts (BGP local/peer IPs, AWS/IBM customer IPs, IP route prefixes), so it now requires IPv4.

Changes:
- New `conversion_test.go`, `helpers_test.go`, `errors_test.go`.
- Extended `vxc_test.go` (ASN boundaries) and `port_test.go` (name-length boundaries).
- Tightened `ValidateIPv4` and `ValidateCIDR` in `helpers.go`; both fixes kept as regression tests.

### CODE REVIEW

File: internal/validation/vxc_test.go [hunk 1]
```diff
@@ -62,6 +62,62 @@ func TestValidateBGPConnectionConfig(t *testing.T) {
...
+				LocalIpAddress: "192.168.1.1",
+				PeerIpAddress:  "192.168.1.2",
+			},
+			wantErr: false,
+		},
+		{
+			name: "ASN negative still passes (only zero is gated)",
+			conn: megaport.BgpConnectionConfig{
+				PeerAsn:        -1,
+				LocalIpAddress: "192.168.1.1",
```
[the remark this answers is not in the document]
Comment (Phil-Browne) [reply internal/validation/vxc_test.go:109]: Fixed in 30dbbd1. ValidateBGPConnectionConfig (and ValidateAWSPartnerConfig, which shared the gap) now reject ASNs outside 1-4294967295; the test expects a ValidationError for PeerAsn -1.

File: internal/validation/conversion_test.go [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+		name  string
+		value interface{}
+		want  int
+		ok    bool
+	}{
+		{"int", 42, 42, true},
+		{"negative int", -7, -7, true},
+		{"zero int", 0, 0, true},
+		{"float64 whole", float64(10), 10, true},
+		{"float64 truncates", float64(3.9), 3, true},
```
[the remark this answers is not in the document]
Comment (Phil-Browne) [reply internal/validation/conversion_test.go]: Fixed in 30dbbd1. GetIntFromInterface now rejects non-whole float64 values (and values outside int range) instead of truncating; the 3.9 case now expects ok=false.

File: internal/validation/conversion.go [hunk 3]
```diff
@@ -12,6 +13,11 @@ func GetIntFromInterface(value interface{}) (int, bool) {
 	case int:
 		return v, true
 	case float64:
+		// Reject fractional or out-of-range values rather than silently truncating
+		// JSON-derived numbers (e.g. 3.9 -> 3).
+		if v != math.Trunc(v) || v < math.MinInt64 || v > math.MaxInt64 {
+			return 0, false
+		}
 		return int(v), true
```
[the remark this answers is not in the document]
Comment (Phil-Browne) [reply internal/validation/conversion.go]: Switched the bounds to math.MinInt/math.MaxInt so 32-bit and wasm targets reject overflowing values. Fixed in 0bd1fdb.

File: internal/validation/conversion_test.go [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+		value interface{}
+		want  int
+		ok    bool
+	}{
+		{"int", 42, 42, true},
+		{"negative int", -7, -7, true},
+		{"zero int", 0, 0, true},
+		{"float64 whole", float64(10), 10, true},
+		{"float64 fractional rejected", float64(3.9), 0, false},
+		{"numeric string", "123", 123, true},
```
[the remark this answers is not in the document]
Comment (Phil-Browne) [reply internal/validation/conversion_test.go:23]: Added a case covering float64(math.MaxInt)*2 (above int range, rejected). Fixed in 0bd1fdb.

File: internal/validation/vxc_test.go [hunk 5]
```diff
@@ -62,6 +62,63 @@ func TestValidateBGPConnectionConfig(t *testing.T) {
 			wantErr: true,
 			errText: "Invalid vRouter interface [0] BGP connection [0] peer ASN: <nil> - is required",
 		},
+		// ASN boundary cases. The validator only gates on zero (required);
+		// any other value passes, so each of these is accepted.
```
[the remark this answers is not in the document]
Comment (Phil-Browne) [reply internal/validation/vxc_test.go]: Rewrote the stale comment to reflect the 1-4294967295 ASN range. Fixed in 0bd1fdb.

File: internal/validation/helpers.go [hunk 6]
```diff
@@ -43,21 +44,23 @@ func ValidateIPv4(ip string, fieldName string) error {
...
+// Returns a ValidationError if the string is empty or not a valid IPv4 CIDR.
 func ValidateCIDR(cidr string, fieldName string) error {
 	if cidr == "" {
 		return NewValidationError(fieldName, cidr, "cannot be empty")
 	}
-	_, _, err := net.ParseCIDR(cidr)
-	if err != nil {
+	ip, _, err := net.ParseCIDR(cidr)
+	if err != nil || ip.To4() == nil || strings.Contains(cidr, ":") {
 		return NewValidationError(fieldName, cidr, "must be a valid CIDR notation")
```
[the remark this answers is not in the document]
Comment (Phil-Browne) [reply internal/validation/helpers.go]: Error message now reads "must be a valid IPv4 CIDR notation" and all assertions updated. Fixed in 0bd1fdb.

File: internal/validation/helpers_test.go [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+func TestIsValidationError(t *testing.T) {
+	tests := []struct {
+		name string
+		err  error
+		want bool
+	}{
+		{"validation error", NewValidationError("f", 1, "bad"), true},
+		{"plain error", errors.New("boom"), false},
+		{"nil error", nil, false},
+		{"wrapped validation error not unwrapped", errors.New("wrap: " + NewValidationError("f", 1, "bad").Error()), false},
```
[the remark this answers is not in the document]
Comment (Phil-Browne) [reply internal/validation/helpers_test.go]: Renamed the case to "plain error containing validation message string" to match what it tests. Fixed in 0bd1fdb.

File: internal/validation/vxc.go [hunk 8]
```diff
@@ -511,6 +517,9 @@ func ValidateBGPConnectionConfig(conn megaport.BgpConnectionConfig, ifaceIndex,
 	if conn.PeerAsn == 0 {
 		return NewValidationError(fmt.Sprintf("%s peer ASN", fieldPrefix), nil, "is required")
 	}
+	if conn.PeerAsn < MinASN || int64(conn.PeerAsn) > MaxASN {
+		return NewValidationError(fmt.Sprintf("%s peer ASN", fieldPrefix), conn.PeerAsn, fmt.Sprintf("must be between %d-%d", MinASN, MaxASN))
+	}
```
[the remark this answers is not in the document]
Comment (Phil-Browne) [reply internal/validation/vxc.go:523]: Updated the ValidateBGPConnectionConfig doc comment to state the peer ASN must be within 1-4294967295, and applied the same to the AWS validator comment for consistency. Pushed in d497ce0.

File: internal/validation/vxc_test.go [hunk 9]
```diff
@@ -62,6 +62,73 @@ func TestValidateBGPConnectionConfig(t *testing.T) {
...
+		},
+		{
+			name: "ASN 4294967295 (32-bit max)",
+			conn: megaport.BgpConnectionConfig{
+				PeerAsn:        4294967295,
+				LocalIpAddress: "192.168.1.1",
+				PeerIpAddress:  "192.168.1.2",
+			},
+			wantErr: false,
+		},
```
[the remark this answers is not in the document]
Comment (Phil-Browne) [reply internal/validation/vxc_test.go]: Right, PeerAsn is an int and these constants overflow int on 32-bit/wasm. Moved the two 32-bit-max boundary cases out of the constant table into TestValidateBGPConnectionConfig_HighASN, which builds the values from a variable (runtime conversion, no compile-time overflow) and skips when int is too narrow. Verified the test package now compiles under GOOS=js GOARCH=wasm. Pushed in bf53760.

File: internal/validation/vxc_test.go [hunk 10]
```diff
@@ -62,6 +62,73 @@ func TestValidateBGPConnectionConfig(t *testing.T) {
...
+		{
+			name: "ASN above 32-bit max rejected",
+			conn: megaport.BgpConnectionConfig{
+				PeerAsn:        4294967296,
+				LocalIpAddress: "192.168.1.1",
+				PeerIpAddress:  "192.168.1.2",
+			},
+			wantErr: true,
+			errText: "Invalid vRouter interface [0] BGP connection [0] peer ASN: 4294967296 - must be between 1-4294967295",
+		},
```
[the remark this answers is not in the document]
Comment (Phil-Browne) [reply internal/validation/vxc_test.go]: Same fix: this above-max case moved into the runtime-constructed, 32-bit-guarded TestValidateBGPConnectionConfig_HighASN. Pushed in bf53760.

File: internal/validation/vxc.go [hunk 11]
```diff
@@ -187,6 +190,7 @@ func ValidateVXCRequest(req *megaport.BuyVXCRequest) error {
 // Validation checks include:
 //   - Connect type must be provided and be one of the valid types ('AWS', 'AWSHC', 'private', 'public')
 //   - Owner account must be provided (AWS account ID)
+//   - ASN must be provided and within the valid range (1-4294967295)
 //   - If customer IP address is provided, it must be in valid CIDR notation
 //   - If Amazon IP address is provided, it must be in valid CIDR notation
```
[the remark this answers is not in the document]
Comment (Phil-Browne) [reply internal/validation/vxc.go]: Updated to "valid IPv4 CIDR notation" for the AWS customer and Amazon IP bullets, matching ValidateCIDR being IPv4-only. Applied the same to the other IPv4-only CIDR bullets (IBM, vRouter, NAT, route prefix) for consistency. Pushed in fb596ff.

File: internal/validation/vxc.go [hunk 12]
```diff
@@ -495,7 +502,7 @@ func ValidateVXCPartnerConfig(config megaport.VXCPartnerConfiguration) error {
 //   - connIndex: The index of this BGP connection within the interface (used for error messages)
 //
 // Validation checks include:
-//   - Peer ASN must be provided (non-zero)
+//   - Peer ASN must be provided and within the valid range (1-4294967295)
 //   - Local IP address must be provided and be a valid IPv4 address or CIDR
 //   - Peer IP address must be provided and be a valid IPv4 address or CIDR
```
[the remark this answers is not in the document]
Comment (Phil-Browne) [reply internal/validation/vxc.go]: Updated the local/peer IP bullets to "valid IPv4 address or IPv4 CIDR". Pushed in fb596ff.

**Review by penzeliz-megaport — APPROVED** [review]
Strong validation coverage with real boundary-value cases and exact error-message assertions. Worth noting for other reviewers that this isn't purely test code: it also tightens production validation (ASN range check, IPv4/CIDR now rejecting IPv4-mapped IPv6 forms, GetIntFromInterface rejecting fractional/overflowing JSON numbers). Those changes are correct, and the 32-bit/js-wasm handling (runtime conversion + skip guard for the max-boundary cases) is carefully done. Signing off the behaviour change along with the tests. LGTM.
